firebase-ml-modeldownloader/src/test/java/com/google/firebase/ml/modeldownloader/FirebaseModelDownloaderTest.java

```diff
@@ -14,34 +14,71 @@
 
 package com.google.firebase.ml.modeldownloader;
 
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertThrows;
+import static org.mockito.Mockito.when;
 
 import androidx.test.core.app.ApplicationProvider;
+import com.google.android.gms.tasks.Task;
 import com.google.firebase.FirebaseApp;
 import com.google.firebase.FirebaseOptions;
+import com.google.firebase.FirebaseOptions.Builder;
+import com.google.firebase.ml.modeldownloader.internal.SharedPreferencesUtil;
+import java.util.Collections;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import org.junit.After;
 import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
 import org.robolectric.RobolectricTestRunner;
 
 @RunWith(RobolectricTestRunner.class)
 public class FirebaseModelDownloaderTest {
 
   public static final String TEST_PROJECT_ID = "777777777777";
+  public static final FirebaseOptions FIREBASE_OPTIONS =
+      new Builder()
+          .setApplicationId("1:123456789:android:abcdef")
+          .setProjectId(TEST_PROJECT_ID)
+          .build();
   public static final String MODEL_NAME = "MODEL_NAME_1";
   public static final CustomModelDownloadConditions DEFAULT_DOWNLOAD_CONDITIONS =
       new CustomModelDownloadConditions.Builder().build();
 
+  public static final String MODEL_HASH = "dsf324";
+  // TODO replace with uploaded model.
+  CustomModel CUSTOM_MODEL = new CustomModel(MODEL_NAME, 0, 100, MODEL_HASH);
+
+  FirebaseModelDownloader firebaseModelDownloader;
+  @Mock SharedPreferencesUtil mockPrefs;
+
+  ExecutorService executor;
+
   @Before
   public void setUp() {
+    MockitoAnnotations.initMocks(this);
     FirebaseApp.clearInstancesForTest();
     // default app
-    FirebaseApp.initializeApp(
-        ApplicationProvider.getApplicationContext(),
-        new FirebaseOptions.Builder()
-            .setApplicationId("1:123456789:android:abcdef")
-            .setProjectId(TEST_PROJECT_ID)
-            .build());
+    FirebaseApp.initializeApp(ApplicationProvider.getApplicationContext(), FIREBASE_OPTIONS);
+
+    executor = Executors.newSingleThreadExecutor();
+    firebaseModelDownloader = new FirebaseModelDownloader(FIREBASE_OPTIONS, mockPrefs, executor);
+  }
+
+  @After
+  public void cleanUp() {
+    try {
```

Comment:
AFAIK this is no longer needed when using the single thread executor.

Reply:
thanks - going too fast.